lib/features/debug/presentation/screens/debug/debug_wm.dart

```diff
@@ -1,162 +1,123 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_template/common/mixin/localization_mixin.dart';
 import 'package:flutter_template/config/url.dart';
 import 'package:flutter_template/features/app/di/app_scope.dart';
 import 'package:flutter_template/features/debug/di/debug_scope.dart';
 import 'package:flutter_template/features/debug/presentation/screens/debug/debug_model.dart';
 import 'package:flutter_template/features/debug/presentation/screens/debug/debug_screen.dart';
 import 'package:flutter_template/features/navigation/service/router.dart';
+import 'package:flutter_template/l10n/app_localizations_x.dart';
 import 'package:provider/provider.dart';
 
-/// Factory for [DebugScreenWidgetModel].
-DebugScreenWidgetModel debugScreenWidgetModelFactory(BuildContext context) {
+/// Factory for [DebugScreenWM].
+DebugScreenWM debugScreenWMFactory(BuildContext context) {
   final appScope = context.read<IAppScope>();
   final debugScope = context.read<IDebugScope>();
   final appRouter = context.read<AppRouter>();
+  final appConfig = appScope.appConfig;
 
   final model = DebugScreenModel(
-    debugScope.debugRepository,
-    appScope.appConfig,
-    appScope.themeService,
+    debugRepository: debugScope.debugRepository,
     logWriter: appScope.logger,
+    serverUrl: appConfig.url,
   );
 
-  return DebugScreenWidgetModel(model, appRouter);
+  return DebugScreenWM(model, router: appRouter, proxyUrl: appConfig.proxyUrl);
+}
+
+/// Interface for [DebugScreenWM].
+abstract class IDebugScreenWM with ILocalizationMixin implements IWidgetModel {
+  /// Text editing controller for proxy url.
+  TextEditingController get proxyEditingController;
+
+  /// Listener current server url state.
+  ValueListenable<Url> get serverUrlState;
+
+  /// Callback of clicking on the url radio button.
+  void onUrlRadioButtonPressed(Url? url);
+
+  /// Сallback of pressed on the server change button.
+  void onChangeServerPressed(Url url);
+
+  /// Сallback of pressed on the connect proxy button.
+  void onConnectProxyPressed();
+
+  /// Set theme mode for app.
+  void onSetThemeMode(ThemeMode? themeMode);
+
+  /// Navigate to ui kit screen.
+  void openUiKit();
 }
 
 /// Widget Model for [DebugScreen].
-class DebugScreenWidgetModel extends WidgetModel<DebugScreen, DebugScreenModel> implements IDebugScreenWidgetModel {
-  /// Empty string.
-  static const String _emptyString = '';
+class DebugScreenWM extends WidgetModel<DebugScreen, DebugScreenModel>
+    with LocalizationMixin
+    implements IDebugScreenWM {
+  final AppRouter _router;
 
-  /// Class that coordinates navigation for the whole app.
-  final AppRouter router;
+  final String? _initialProxyUrl;
 
-  late final _textEditingController = TextEditingController(text: model.proxyUrl);
+  late final TextEditingController _textEditingController;
 
-  @override
-  late final ValueNotifier<Url> urlState;
+  late final ScaffoldMessengerState _scaffoldMessenger;
```

Comment:
I don't think that we need `late final` variable. We can inject `ScaffoldMessengerState` through `DebugScreenWM` constructor.